subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/tasks/XcTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest.tasks;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.file.Directory;
+import org.gradle.api.file.DirectoryVar;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.testing.Test;
+import org.gradle.nativeplatform.test.xctest.internal.NativeTestExecuter;
+
+import javax.inject.Inject;
+import java.io.File;
+
+/**
+ * Executes XCTest tests. Test are always run in a single execution.
+ *
+ * @since 4.2
+ */
+@Incubating
+public class XcTest extends Test {
```

Comment:
I will write a design doc for this, see https://github.com/gradle/gradle-native/issues/93